### ISSUE TITLE
VotingOnlyNodeCoordinatorTests testDoesNotElectVotingOnlyMasterNode failure

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#43631]
https://scans.gradle.com/s/tfu25vregk2hy
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=fedora-29&&immutable/86/console

```
java.lang.AssertionError: (No message provided)
at __randomizedtesting.SeedInfo.seed([D28433B53AFFA3C7:4E4230F5B099413]:0)
at org.junit.Assert.fail(Assert.java:86)
at org.junit.Assert.assertTrue(Assert.java:41)
at org.junit.Assert.assertFalse(Assert.java:64)
at org.junit.Assert.assertFalse(Assert.java:74)
at org.elasticsearch.cluster.coordination.VotingOnlyNodeCoordinatorTests.testDoesNotElectVotingOnlyMasterNode(VotingOnlyNodeCoordinatorTests.java:29)
```

Log file: https://scans.gradle.com/s/tfu25vregk2hy/tests/tgrx2aitn6u62-vhkxezy67i4ic

This actually reproduces for me:
```
./gradlew :x-pack:plugin:voting-only-node:test --tests "org.elasticsearch.cluster.coordination.VotingOnlyNodeCoordinatorTests.testDoesNotElectVotingOnlyMasterNode" -Dtests.seed=D28433B53AFFA3C7 -Dtests.security.manager=true -Dtests.locale=sv-AX -Dtests.timezone=Asia/Dhaka -Dcompiler.java=12 -Druntime.java=11
```

**Comment by davidkyle** [comment]
There have been a handful of failures on master in recent days. Muted on master in fc5ce18ae611744ea8c19bf88bd1d03f748672ac

**Comment by ywelsch** [comment]
Will be fixed by #43628

**Comment by ywelsch** [comment]
Closed by #43628

**Comment by mayya-sharipova** [comment]
Looks like this test still fails on 7.3. Another failure today, which reproduces for me:

```
./gradlew :x-pack:plugin:voting-only-node:test --tests "org.elasticsearch.cluster.coordination.VotingOnlyNodeCoordinatorTests.testDoesNotElectVotingOnlyMasterNode" \
  -Dtests.seed=5C6125B8EF7A9BC \
  -Dtests.security.manager=true \
  -Dtests.locale=sr-RS \
  -Dtests.timezone=Etc/Zulu \
  -Dcompiler.java=12 \
  -Druntime.java=8
```
```
java.lang.AssertionError: node0 has the same applied state as node1
    Expected: <13L>
         but: was <0L>
        at __randomizedtesting.SeedInfo.seed([5C6125B8EF7A9BC:D3A602E1EF019E68]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.stabilise(AbstractCoordinatorTestCase.java:509)
        at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.stabilise(AbstractCoordinatorTestCase.java:463)
        at org.elasticsearch.cluster.coordination.VotingOnlyNodeCoordinatorTests.testDoesNotElectVotingOnlyMasterNode(VotingOnlyNodeCoordinatorTests.java:53)
```
Log : https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.3+intake/6/console

Build Scans: https://gradle.com/s/yqiidnchqxlsk
